augur/generate.py

```diff
@@ -0,0 +1,266 @@
+
+    Parameters:
+    ----------
+    config : dict
+        The dictinary containt the relevant two_point section of the config
+
+    Returns:
+    -------
+    sacc : sacc object
+
+       Sacc objects with appropriate tracers and measurement slots
+       (i.e. data vectors with associated correlation functions,
+       angles, etc), but with zeros for measurement values
+
+    """
+
+    S = sacc.Sacc()
+    verbose = config['verbose']
+    if verbose:
+        print("Generating tracers: ", end="")
+    for src, tcfg in config['sources'].items():
+        if verbose:
+            print(src, " ", end="")
+        if "Nz_type" not in tcfg:
+            print("Missing Nz_type in %s. Quitting." % src)
+            raise RuntimeError
+        if tcfg['Nz_type'] == 'Gaussian':
+            mu, sig = tcfg['Nz_center'], tcfg['Nz_sigma']
+            zar = np.linspace(max(0, mu - 5 * sig), mu + 5 * sig, 500)
+            Nz = np.exp(-(zar - mu)**2 / (2 * sig**2))
+            S.add_tracer('NZ', src, zar, Nz)
+        elif tcfg['Nz_type'] == 'TopHat':
+            mu, wi = tcfg['Nz_center'], tcfg['Nz_width']
+            zar = np.linspace(max(0, mu - wi / 2), mu + wi / 2, 5)
+            Nz = np.ones(5)
+            S.add_tracer('NZ', src, zar, Nz)
+        else:
+            print("Bad Nz_type in %s. Quitting." % src)
+            raise RuntimeError
+    if verbose:
+        print("\nGenerating data slots: ", end="")
+    for name, scfg in config['statistics'].items():
+        if verbose:
+            print(name, " ", end="")
+        dt = scfg['sacc_data_type']
+        src1, src2 = scfg['sources']
+        if 'cl' in dt:
+            ell_edges = np.array(scfg['ell_edges'])
+            ells = 0.5 * (ell_edges[:-1] + ell_edges[1:])
+            for ell in ells:
+                S.add_data_point(dt, (src1, src2), 0.0, ell=ell, error=1e30)
+        elif 'xi' in dt:
+            theta_edges = np.array(scfg['theta_edges'])
+            thetas = 0.5 * (theta_edges[:-1] + theta_edges[1:])
+            for theta in thetas:
+                S.add_data_point(dt, (src1, src2), 0.0,
+                                 theta=theta, error=1e30)
+        else:
+            print("Cannot process %s. Quitting." % dt)
+            raise NotImplementedError
+
+    if verbose:
+        print()
+    config['sacc_data'] = S
+
+
+def getNoisePower(config, src):
+    """ Returns noise power for tracer
+
+    Parameters:
+    ----------
+    config : dict
+        The dictinary containt the relevant two_point section of the config
+
+    src : str
+        Tracer ID
+
+    Returns:
+    -------
+    noise : float
+       Noise power for Cls for that particular tracer. That is 1/nbar for
+       number counts and e**2/nbar for weak lensing tracer
+
+    """
+
+    d = config['sources'][src]
+    nbar = d['number_density'] * (180 * 60 / np.pi)**2  # per steradian
+    kind = d['kind']
+    if kind == 'WLSource':
+        NPower = d['ellipticity_error']**2 / nbar
```

Comment:
python variable names are traditionally lower-case and snake-case. Classes are camel-case.